app/javascript/profileDropdown/flagButton.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-alert */
+export function initFlag() {
+  const flagButton = document.getElementById('user-profile-dropdownmenu-flag-button');
+
+  if (!flagButton) {
+    // button not always present when this is called
+    return;
+  }
+
+  /* eslint-disable-next-line no-undef */
+  const user = userData();
+  if (!user) {
+    return;
+  }
+
+  const { profileUserId, profileUserName } = flagButton.dataset;
+  let flagStatus = flagButton.dataset.flagStatus === 'true';
+
+  if (user.id === parseInt(profileUserId, 10) || !user.trusted) {
+    flagButton.remove();
+  }
+
+  function flag() {
+    const confirmFlag = window.confirm(
+      flagStatus
+        ? "Are you sure you want to unflag this person? This will restore their posts' visibility."
+        : 'Are you sure you want to flag this person? This will make all of their posts less visible.',
+    );
```

Comment:
I thought about this while QAing this and then went back to the original RFC to look at the [definition of done](https://github.com/forem/rfcs/pull/73/files#diff-5dc7721fc901e825ee9a52b83cb6c4cb07c72c6df4286f97fbdb76fdba4f6408R31-R33)...

It feels a bit weird to me to flag a user and have to click through _two_ confirmation buttons. I know that @lisasy mentioned in [this comment](https://github.com/forem/forem/pull/12862#issuecomment-796875917) that we the mod should receive _"feedback that this [the flagging] successfully happened"_, but I think we might want to make this a more _passive_ notice rather than an "active" one where the mod has to click twice to flag someone. The first click (the actual "confirmation") makes sense to be more active, as we definitely want to ensure that the mod knows the implication of what they're about to do.

But after that, there isn't any _action_ to be taken, so the browser alert feels kind of heavyweight for this:
<img width="442" alt="Screen Shot 2021-03-26 at 11 34 56 AM" src="https://user-images.githubusercontent.com/6921610/112677845-7273ea00-8e27-11eb-8121-b5928719f157.png">

This is just the feeling I got when QAing the flagging feature; I realize that the RFC doesn't specify what _kind_ of confirmation should be used here 😓 Could we maybe just show the second confirmation as a flash message of sorts (something that doesn't require another click from the mod?)?

@lisasy What do you think?

Reply:
Right, I'm not a big fan of the second confirmation either. I'm not sure how we usually do flash messages in async requests from the frontend, but I can have a look.

Reply:
I just removed the second popup now. The menu item in the existing popup will immediately change from `Flag` to `Unflag` and vice versa, and while potentially a bit too subtle I think it may be enough. Definitely better than the extra popup.